.github/workflows/test.yml

```diff
@@ -122,40 +145,97 @@ jobs:
 
+      - name: (windows) install dxc
+        # from wgpu repo: https://github.com/gfx-rs/wgpu/blob/trunk/.github/workflows/ci.yml
+        if: runner.os == 'Windows'
+        shell: bash
+        run: |
+          set -e
+
+          curl.exe -L --retry 5 https://github.com/microsoft/DirectXShaderCompiler/releases/download/$DXC_RELEASE/$DXC_FILENAME -o dxc.zip
+          7z.exe e dxc.zip -odxc bin/x64/{dxc.exe,dxcompiler.dll,dxil.dll}
+
+          # We need to use cygpath to convert PWD to a windows path as we're using bash.
+          cygpath --windows "$PWD/dxc" >> "$GITHUB_PATH"
+
       - name: (windows) install warp
+        # from wgpu repo: https://github.com/gfx-rs/wgpu/blob/trunk/.github/workflows/ci.yml
         if: runner.os == 'Windows'
         shell: bash
         run: |
           set -e
 
-          curl.exe -L https://www.nuget.org/api/v2/package/Microsoft.Direct3D.WARP/1.0.7.1 -o warp.zip
+          # Make sure dxc is in path.
+          dxc --version
+
+          curl.exe -L --retry 5 https://www.nuget.org/api/v2/package/Microsoft.Direct3D.WARP/$WARP_VERSION -o warp.zip
           7z.exe e warp.zip -owarp build/native/amd64/d3d10warp.dll
 
-          mkdir -p target/debug/deps
+          mkdir -p target/llvm-cov-target/debug/deps
 
-          cp -v warp/d3d10warp.dll target/debug/
-          cp -v warp/d3d10warp.dll target/debug/deps
+          cp -v warp/d3d10warp.dll target/llvm-cov-target/debug/
+          cp -v warp/d3d10warp.dll target/llvm-cov-target/debug/deps
 
       - name: (windows) install mesa
+        # from wgpu repo: https://github.com/gfx-rs/wgpu/blob/trunk/.github/workflows/ci.yml
         if: runner.os == 'Windows'
         shell: bash
         run: |
           set -e
 
-          curl.exe -L https://github.com/pal1000/mesa-dist-win/releases/download/23.2.1/mesa3d-23.2.1-release-msvc.7z -o mesa.7z
+          curl.exe -L --retry 5 https://github.com/pal1000/mesa-dist-win/releases/download/$MESA_VERSION/mesa3d-$MESA_VERSION-release-msvc.7z -o mesa.7z
           7z.exe e mesa.7z -omesa x64/{opengl32.dll,libgallium_wgl.dll,libglapi.dll,vulkan_lvp.dll,lvp_icd.x86_64.json}
 
-          mkdir -p target/debug/deps
+          cp -v mesa/* target/llvm-cov-target/debug/
+          cp -v mesa/* target/llvm-cov-target/debug/deps
 
-          cp -v mesa/* target/debug/
-          cp -v mesa/* target/debug/deps
-
-          echo "VK_DRIVER_FILES=$PWD/mesa/lvp_icd.x86_64.json" >> "$GITHUB_ENV"
+          # We need to use cygpath to convert PWD to a windows path as we're using bash.
+          echo "VK_DRIVER_FILES=`cygpath --windows $PWD/mesa/lvp_icd.x86_64.json`" >> "$GITHUB_ENV"
           echo "GALLIUM_DRIVER=llvmpipe" >> "$GITHUB_ENV"
 
-      - name: (windows) install directx
-        if: runner.os == 'Windows'
-        uses: napokue/setup-dxc@v1.1.0
+      - name: (linux) install vulkan sdk
+        # from wgpu repo: https://github.com/gfx-rs/wgpu/blob/trunk/.github/workflows/ci.yml
```

Comment:
I think we should cite them in NOTICES.md, just making clear that we get inspired by their CI 😅

Reply:
Done